src/datatype.jl

```diff
@@ -0,0 +1,36 @@
+# custom data tyoe to hold ImageHDU
+struct CCDData{T,M<:AbstractMatrix{T}} <: AbstractMatrix{T}
+    data::M
+    hdr::FITSHeader
+end
+
+# extending the AbstractMatrix interface
+Base.size(A::CCDData) = size(A.data)
+Base.getindex(A::CCDData, i) = getindex(A.data, i)
+Base.setindex!(A::CCDData, v, i) = setindex!(A.data, v, i)
```

Comment:
The `*index*` functions have many methods, not just with a single index

Reply:
Are you referring to this one?
```
setindex!(A, v, I::Vararg{Int, N})
```

Reply:
I'd do
```suggestion
Base.getindex(A::CCDData, inds...) = getindex(A.data, inds...)
Base.setindex!(A::CCDData, v, inds...) = setindex!(A.data, v, inds...)
```